postgres/bin/dump-to-s3.sh

```diff
@@ -1,19 +1,28 @@
 #!/bin/bash
 # Usage:  dump-to-s3.sh <s3://...dump> [dbname]
-# Expects a DATABASE_URL environment variable that is the DB to dump
-# Optionally, a dbname can be supplied as the second argument
-# to override the name from the DATABASE_URL
+# Expects a DATABASE_URL environment variable and SERVER_VERSION exported by the entrypoint.
+# Optionally, a dbname can be supplied as the second argument to override the name from DATABASE_URL.
+
 set -euf -o pipefail
 
 cleanup() { rv=$?; if [ -f /tmp/db.dump ]; then shred -u /tmp/db.dump; fi; exit $rv; }
 trap cleanup EXIT
 
-NAME=${2:-$NAME}
-CONNECT_DB_URL="postgres://$USER@$HOST:$PORT/$NAME"
+# Extract database name from DATABASE_URL if not provided as an argument
+DBNAME=${2:-$(echo "$DATABASE_URL" | sed -E 's|.*/([^/?]+).*|\1|')}
+CONNECT_DB_URL="${DATABASE_URL%/*}/$DBNAME"
+
+# Use SERVER_VERSION set by the entrypoint
+PG_DUMP="pg_dump-$SERVER_VERSION"
```

Comment:
Can you make this work if `$SERVER_VERSION` is not supplied? It can default to the latest version.